### PR TITLE
DE-2997: minor enhancements

### PR DESCRIPTION
**Pull request by setiat**
1. Update enforce_properties to list missing or extra propeties.

2. typo fix in datacubes.

### CODE REVIEW

**Review by mlipschultz — APPROVED** [review]

File: ids_validator/checks/rules_checker.py [hunk 1]
```diff
@@ -114,12 +114,24 @@ def enforce_properties(cls, node: Node, properties: list):
...
-        node_properties = list(node_properties.keys())
-        if sorted(node_properties) != sorted(properties):
+        node_properties = node_properties.keys()
+        properties = set(properties)
+
+        extra_properties = node_properties - properties
+        if extra_properties:
             logs += [(
-                f"'properties' must only contain {properties}",
+                f"Extra properties found: {extra_properties}",
```
Comment (wkopen) [inline ids_validator/checks/rules_checker.py:123]: I kinda still like to keep the original message. Just add this "Extra..." after that.

File: ids_validator/checks/rules_checker.py [hunk 2]
```diff
@@ -114,12 +114,24 @@ def enforce_properties(cls, node: Node, properties: list):
...
             )]
+            return logs
+
+        missing_properties = properties - node_properties
+        if missing_properties:
+            logs += [(
+                f"Missing properties: {missing_properties}",
+                Log.CRITICAL.value
+            )]
+            return logs
```
Comment (wkopen) [inline ids_validator/checks/rules_checker.py:134]: Nice add! Can you also add the `f"'properties' must only contain {properties}"` in the beginning so people know what should be there?

**Review by wkopen — COMMENTED** [review]
Two minor things then you are good to merge!

File: ids_validator/checks/rules_checker.py [hunk 3]
```diff
@@ -114,12 +114,24 @@ def enforce_properties(cls, node: Node, properties: list):
...
+        properties = set(properties)
+
+        extra_properties = node_properties - properties
+        if extra_properties:
             logs += [(
-                f"'properties' must only contain {properties}",
+                f"Extra properties found: {extra_properties}",
                 Log.CRITICAL.value
             )]
+            return logs
```
Comment (mlipschultz) [inline ids_validator/checks/rules_checker.py:126]: Currently the code only reports extra properties found or missing properties.  I think a solution to this is to remove the `return logs` from here and from the `missing_properties` block below, then just let the final `return logs` return the logs.

Comment (setiat) [reply ids_validator/checks/rules_checker.py]: fixed. added test cases

File: ids_validator/checks/rules_checker.py (hunk 2 above)
Comment (setiat) [reply ids_validator/checks/rules_checker.py]: fixed

**Review by mlipschultz — APPROVED** [review]
Everything looks good to me!

**Pull request merged**